packages/block-library/src/image/editor.scss

```diff
@@ -28,6 +28,7 @@
 
 .editor-block-list__block[data-type="core/image"][data-align="center"] {
 	.wp-block-image {
+		clear: both;
```

Comment:
Is this needed only on `editor`?

Reply:
Yes, forgot to mention this in the change notes. 

The issue I was seeing was floating a small image left, then centering a small image after that. This would stack up next to the small floated image in the editor, but not the front-end. I did not dig deeper, but I can.